tests/test_gradient_detaching.py

```diff
@@ -0,0 +1,136 @@
+import numpy as np
+import pytest
+import torch
+import xarray as xr
+
+from ocean_emulators.config import SamudraConfig, UNetBackboneConfig
+from ocean_emulators.constants import TensorMap
+from ocean_emulators.datasets import TrainData
+from ocean_emulators.utils.data import DataSource, Normalize
+from ocean_emulators.utils.multiton import MultitonScope
+
+
+@pytest.fixture(params=[0, 1, 2])
+def gradient_detach_interval(request):
+    """Parametrized fixture for gradient detach intervals. `0` means no detaching."""
+    return request.param
```

Comment:
🐑 Looking at how this fixture is used, here's an idea: what if this returned tuples of the interval + the "name" of the interval, i.e. if the gradient is detached or not. It could look like this:
```suggestion
    return (i := request.param), "no detaching" if i == 0 else f"detaching every {i}"
```

I am not sure if this is a good fit, but I bet it would clean up some of the logic in your tests code.

Reply:
this was helpful!